inference-engine/tests/functional/plugin/shared/src/single_layer_tests/loop.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <memory>
+#include <functional>
+
+#include "ie_core.hpp"
+
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "functional_test_utils/precision_utils.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+
+#include "single_layer_tests/loop.hpp"
+
+namespace LayerTestsDefinitions {
+
+    std::string LoopTest::getTestCaseName(const testing::TestParamInfo<LoopParams> &obj) {
+        bool execute_first_iteration;
+        bool is_body_condition_const;
+        bool body_condition; // works only if is_body_condition_const ==
+        int64_t trip_count;
+        std::vector<std::pair<std::vector<size_t>, LOOP_IN_TYPE>> inputs;
+        InferenceEngine::Precision netPrecision;
+        std::string targetDevice;
+        std::tie(execute_first_iteration, is_body_condition_const, body_condition, trip_count, inputs, netPrecision,
+                 targetDevice) = obj.param;
+
+        std::vector<std::vector<size_t>> inputs_separate;
+        std::vector<LOOP_IN_TYPE> types_separate;
+        for (auto &el : inputs) {
+            inputs_separate.push_back(el.first);
+            types_separate.push_back(el.second);
+        }
+        std::ostringstream result;
+        result << "execute_first_iteration" << execute_first_iteration << "_";
+        result << "is_body_condition_const=" << is_body_condition_const << "_";
+        result << "body_condition=" << body_condition << "_";
+        result << "trip_count=" << trip_count << "_";
+        result << "IS=" << CommonTestUtils::vec2str(inputs_separate) << "_";
+        result << "types=" << CommonTestUtils::vec2str(types_separate) << "_";
+        result << "netPRC=" << netPrecision.name() << "_";
+        result << "targetDevice=" << targetDevice << "_";
+        return result.str();
+    }
+
+    void LoopTest::SetUp() {
+        SKIP_IF_CURRENT_TEST_IS_DISABLED()
+        SetRefMode(LayerTestsUtils::IE);
+        bool execute_first_iteration;
+        bool is_body_condition_const;
+        bool body_condition; // works only if is_body_condition_const ==
+        int64_t trip_count;
+        std::vector<std::pair<std::vector<size_t>, LOOP_IN_TYPE>> inputs;
+        InferenceEngine::Precision netPrecision;
+        std::tie(execute_first_iteration, is_body_condition_const, body_condition, trip_count, inputs, netPrecision,
+                 targetDevice) = this->GetParam();
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+        // That which we iterate over
+        std::vector<std::vector<size_t>> inputs_separate;
```

Comment:
may be, we should move to the builders this large part of code?

Reply:
It is not yet clear how to do this if we want to test different ngraph functions inside the Loop and it's just a prototype, now these tests are disabled.
They will be enabled and finalized in the next PR